### ISSUE TITLE
Flixel doing nothing to clear memory footprint

### ISSUE THREAD

**Issue by vonWolfehaus** [AdamAtomic/flixel#122]
This problem often comes up in the forums from time to time. The root of the problem is that most (all?) destroy methods in Flixel are EMPTY. All variables need to be nulled out and bitmapdata disposed of in these functions.

I wrote a post that details this problem and I provide a couple replacement functions for FlxObject and FlxSprite as a start. View it here: http://coldconstructs.com/2011/03/memory-management-in-flixel/

**Comment by alteveer** [comment]
how about forking and patching?

**Comment by vonWolfehaus** [comment]
I don't know how yet. Should I figure it out first or can I just post a zip with the mods?

**Comment by alteveer** [comment]
github has a great guides and pointers here: http://help.github.com/. once you get git installed, you can 
1. start by pressing the "fork" button near the top. 
2. then, clone the repo locally and apply your changes 
3. test, 
4. push back to origin.

At this point, you should have the changes in your own repo on github. Do a pull request, and hope that Mr. Atomic wants to include it in his repo. regardless, that change is available for me to merge into my repo, if I want.

**Comment by vonWolfehaus** [comment]
Holy crap that's awesome! Thanks so much man, I'll get on that next chance I get.

**Comment by vonWolfehaus** [comment]
Thanks to you I got started and forked/patched it and did a pull request. https://github.com/vonWolfehaus/flixel Prost!

**Comment by alteveer** [comment]
awesome. i will pull it into my branch and check it out. Can you explain the uint -> int changes? what does this do?

**Comment by alteveer** [comment]
Also, do you have benchmarks/examples of the wins here?

**Comment by vonWolfehaus** [comment]
getTimer() returns int, so casting it into uint is completely unnecessary. Better safe than sorry. uint is also a little slower to work with, but it keeps getting faster with each new Flash Player release.

I don't have a deep memory profiler--I just observed changes using the new Stats I added to Flixel. Before, it maxed at 32 and looped from 17-28. After my changes, it maxed at 27 and looped through 14-24, and changing states didn't have nearly a high of an impact as before. Pretty huge considering I just nulled stuff out, but your mileage may vary. I used FlxCollisions as the testbed.
